mod-audit-server/src/main/java/org/folio/services/marc/impl/MarcAuditServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.services.marc.impl;
+
+import io.vertx.core.Future;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.marc.MarcAuditDao;
+import org.folio.dao.marc.MarcAuditEntity;
+import org.folio.services.marc.MarcAuditService;
+import org.folio.util.marc.ParsedRecordUtil;
+import org.folio.util.marc.SourceRecordDomainEvent;
+import org.springframework.stereotype.Service;
+
+import static org.folio.util.ErrorUtils.handleFailures;
+
+@Service
+public class MarcAuditServiceImpl implements MarcAuditService {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private final MarcAuditDao marcAuditDao;
+
+  public MarcAuditServiceImpl(MarcAuditDao marcAuditDao) {
+    this.marcAuditDao = marcAuditDao;
+  }
+
+  @Override
+  public Future<RowSet<Row>> saveMarcDomainEvent(SourceRecordDomainEvent event) {
+    var tenantId = event.getEventMetadata().getTenantId();
+    LOGGER.debug("saveOrderAuditEvent:: Trying to save SourceRecordDomainEvent tenantId: '{}', eventId: '{};", tenantId, event.getEventId());
+    MarcAuditEntity entity;
+    try {
+      entity = ParsedRecordUtil.mapToEntity(event);
+    } catch (Exception e) {
+      LOGGER.warn("saveMarcBibDomainEvent:: Error during mapping SourceRecordDomainEvent to MarcAuditEntity for event '{}'", event.getEventId());
+      return handleFailures(e, event.getEventId());
+    }
+    if (entity.diff() == null || entity.diff().isEmpty()) {
+      LOGGER.debug("saveMarcBibDomainEvent:: No changes detected, skipping save record '{}' and tenantId='{}'", entity.recordId(), tenantId);
+      return Future.succeededFuture();
+    }
+    return marcAuditDao.save(entity, tenantId)
+      .recover(throwable -> {
+        LOGGER.error("handleFailures:: Could not save order audit event for tenantId: {}", tenantId);
```

Comment:
@dmytrokrutii  I hope here in log should be still `saveMarcDomainEvent` method name for better logs investigation in future.

Reply:
> @dmytrokrutii Also let's try to add some information about the entity here. I mean Id, or the whole entity. What do you think?

Depending on the type of event, we need to get this data from different places (`newRecord` or `old`), which is a bit inconvenient for logging. After successfully mapping in `MarcAuditEntity`, we can log information about the entity. I can add the `eventId` parameter to the logs in DAO.
Or there is a need to change the structure of the SourceRecordDomainEvent and the logic of the buildSourceRecordDomainEvent method.

Reply:
@dmytrokrutii Yes, let's add eventId